### ISSUE TITLE
$a[$#a--] does not give last element

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#8967]
Migrated from [rt.perl.org#43705](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=43705) (status was 'resolved')

Searchable as RT43705$

**Comment by p5pRT** [comment]
### From perlbug@schmorp.de

#### Created by perlbug@schmorp.de



perldoc \-f pop says&#8203;:

&nbsp;   Has an effect similar to $ARRAY\[$\#ARRAY\-\-\]

but it doesn't work&#8203;:

&nbsp;   perl \-e '@&#8203;a=\(4\,5\,6\); warn $a\[$\#a\-\-\]'
&nbsp;   Warning&#8203;: something's wrong at \-e line 1\.
&nbsp;   4 5 at \-e line 1\.

without the \-\-\, it does give me the last element&#8203;:

&nbsp;   perl \-e '@&#8203;a=\(4\,5\,6\); warn $a\[$\#a\]; warn "@&#8203;a"'
&nbsp;   6 at \-e line 1\.
&nbsp;   4 5 6 at \-e line 1\.

\(this bug is of paramount \*un\*importance to me\, it breaks no code known to
me except the one in the documentation\, btw\.\)

<details><summary>Perl Info</summary>

```

Flags:
    category=core
    severity=low

Site configuration information for perl v5.8.8:

Configured by Marc Lehmann at Sat Jan 20 00:23:48 CET 2007.

Summary of my perl5 (revision 5 version 8 subversion 8) configuration:
  Platform:
    osname=linux, osvers=2.6.17.6, archname=amd64-linux
    uname='linux cerebro 2.6.17.6 #1 smp fri oct 20 19:28:13 cest 2006 x86_64 gnulinux '
    config_args='-Duselargefiles -Dxxxxuse64bitint -Uuse64bitall -Dusemymalloc=n -Dcc=gcc -Dccflags=-DPERL_DONT_CREATE_GVSV_disabled -ggdb -Dcppflags=-DPERL_DONT_CREATE_GVSV_disabled -D_GNU_SOURCE -I/opt/include -Doptimize=-O6 -march=opteron -mtune=opteron -funroll-loops -fno-strict-aliasing -Dcccdlflags=-fPIC -Dldflags=-L/opt/perl/lib -L/opt/lib -Dlibs=-ldl -lm -lcrypt -Darchname=amd64-linux -Dprefix=/opt/perl -Dprivlib=/opt/perl/lib/perl5 -Darchlib=/opt/perl/lib/perl5 -Dvendorprefix=/opt/perl -Dvendorlib=/opt/perl/lib/perl5 -Dvendorarch=/opt/perl/lib/perl5 -Dsiteprefix=/opt/perl -Dsitelib=/opt/perl/lib/perl5 -Dsitearch=/opt/perl/lib/perl5 -Dsitebin=/opt/perl/bin -Dman1dir=/opt/perl/man/man1 -Dman3dir=/opt/perl/man/man3 -Dsiteman1dir=/opt/perl/man/man1 -Dsiteman3dir=/opt/perl/man/man3 -Dman1ext=1 -Dman3ext=3 -Dpager=/usr/bin/less -Uafs -Uusesfio -Uusenm -Uuseshrplib -Dd_dosuid -Dusethreads=undef -Duse5005threads=undef -Duseithreads=undef -Dusemultiplicity=undef -Demail=perl-binary@plan9.de -Dcf_email=perl-binary@plan9.de -Dcf_by=Marc Lehmann -Dlocincpth=/opt/perl/include /opt/include -Dmyhostname=localhost -Dmultiarch=undef -Dbin=/opt/perl/bin -Dusedevel -des'
    hint=recommended, useposix=true, d_sigaction=define
    usethreads=undef use5005threads=undef useithreads=undef usemultiplicity=undef
    useperlio=define d_sfio=undef uselargefiles=define usesocks=undef
    use64bitint=define use64bitall=undef uselongdouble=undef
    usemymalloc=n, bincompat5005=undef
  Compiler:
    cc='gcc', ccflags ='-DPERL_DONT_CREATE_GVSV_disabled -ggdb -fno-strict-aliasing -pipe -I/opt/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64',
    optimize='-O6 -march=opteron -mtune=opteron -funroll-loops -fno-strict-aliasing',
    cppflags='-DPERL_DONT_CREATE_GVSV_disabled -D_GNU_SOURCE -I/opt/include -DPERL_DONT_CREATE_GVSV_disabled -ggdb -fno-strict-aliasing -pipe -I/opt/include'
    ccversion='', gccversion='4.1.2 20061115 (prerelease) (Debian 4.1.1-21)', gccosandvers=''
    intsize=4, longsize=8, ptrsize=8, doublesize=8, byteorder=12345678
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=16
    ivtype='long', ivsize=8, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=8
    alignbytes=8, prototype=define
  Linker and Libraries:
    ld='gcc', ldflags ='-L/opt/perl/lib -L/opt/lib -L/usr/local/lib'
    libpth=/usr/local/lib /lib /usr/lib
    libs=-ldl -lm -lcrypt
    perllibs=-ldl -lm -lcrypt
    libc=/lib/libc-2.3.6.so, so=so, useshrplib=false, libperl=libperl.a
    gnulibc_version='2.3.6'
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-Wl,-E'
    cccdlflags='-fPIC', lddlflags='-shared -L/opt/perl/lib -L/opt/lib -L/usr/local/lib'

Locally applied patches:
    MAINT29832


@INC for perl v5.8.8:
    /root/src/sex
    /opt/perl/lib/perl5
    /opt/perl/lib/perl5
    /opt/perl/lib/perl5
    /opt/perl/lib/perl5
    /opt/perl/lib/perl5
    .


Environment for perl v5.8.8:
    HOME=/root
    LANG (unset)
    LANGUAGE (unset)
    LC_CTYPE=de_DE.UTF-8
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/root/s2:/root/s:/opt/bin:/opt/sbin:/bin:/sbin:/usr/bin:/usr/sbin:/usr/X11/bin:/usr/games:/usr/local/bin:/usr/local/sbin:/root/src/uunet:.
    PERL5LIB=/root/src/sex
    PERL5_CPANPLUS_CONFIG=/root/.cpanplus/config
    PERLDB_OPTS=ornaments=0
    PERL_BADLANG (unset)
    PERL_UNICODE=EAL
    SHELL=/bin/bash

```
</details>

**Comment by p5pRT** [comment]
### From @rgs

On 10/07/07\, via RT Marc Lehmann \<perlbug\-followup@&#8203;perl\.org> wrote&#8203;:
> perldoc \-f pop says&#8203;:
>
>    Has an effect similar to $ARRAY\[$\#ARRAY\-\-\]
>
> but it doesn't work&#8203;:
>
>    perl \-e '@&#8203;a=\(4\,5\,6\); warn $a\[$\#a\-\-\]'
>    Warning&#8203;: something's wrong at \-e line 1\.
>    4 5 at \-e line 1\.
>
> without the \-\-\, it does give me the last element&#8203;:
>
>    perl \-e '@&#8203;a=\(4\,5\,6\); warn $a\[$\#a\]; warn "@&#8203;a"'
>    6 at \-e line 1\.
>    4 5 6 at \-e line 1\.
>
> \(this bug is of paramount \*un\*importance to me\, it breaks no code known to
> me except the one in the documentation\, btw\.\)

To me that's a doc bug\. \(using \-MO=Concise\,\-exec explains that the
postdec happens before the array element is taken\, and I don't see how
that could be changed\.\)

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From @rgs

Documentation fixed by change \#31584\.

**Comment by p5pRT** [comment]
### From [Unknown Contact. See original ticket]

Documentation fixed by change \#31584\.

**Comment by p5pRT** [comment]
@rgs - Status changed from 'open' to 'resolved'
